### PR TITLE
Incorporated glog into progen

### PR DESCRIPTION
**Pull request by jakpiase**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,7 +1,12 @@
 #include <iostream>
+#include <glog/logging.h>
 
-int main() {
-    std::cout << "Hello world!\n";
+int main(int argc, char* argv[]) {
+    google::InitGoogleLogging(argv[0]);
+    FLAGS_logtostderr = 1;
```
Comment (wrobcio789) [inline src/main.cpp]:
```suggestion
    FLAGS_logtostderr = true;
```

Clang Tidy

File: src/main.cpp [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+    LOG(INFO) << "Starting ProGen";
 
+
+    google::ShutdownGoogleLogging();
+    LOG(INFO) << "ProGen exited succesfully";
```
Comment (wrobcio789) [inline src/main.cpp:11]: swap these lines

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -1,7 +1,12 @@
-set(BINARY ${CMAKE_PROJECT_NAME})
+set(BINARY ${CMAKE_PROJECT_NAME}_run)
 
 file(GLOB_RECURSE SOURCES LIST_DIRECTORIES true *.h *.cpp)
+list(REMOVE_ITEM SOURCES "main.cpp")
 set(SOURCES ${SOURCES})
 
-add_executable(${BINARY}_run ${SOURCES})
-add_library(${BINARY}_lib STATIC ${SOURCES})
+add_executable(${BINARY} main.cpp)
```
Comment (wrobcio789) [inline src/CMakeLists.txt:7]: I'd suggest removing BINARY variable and writing here ${CMAKE_PROJECT_NAME}_run

Comment (wrobcio789) [reply src/CMakeLists.txt:7]: cause it's only used in one place now

**Review by wrobcio789 — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt (hunk 3 above)
Comment (jakpiase) [reply src/CMakeLists.txt:7]: It's used in two places and will be used in many more when we will have more dependencies.

Comment (wrobcio789) [reply src/CMakeLists.txt:7]: A, ok, true

**Review by wrobcio789 — APPROVED** [review]
Great job on this PR, Jakub! Keep up the good work!

**Pull request merged**